File: app/src/commonMain/kotlin/com/mountaincrab/logrhythm/data/model/Bristol.kt
package com.mountaincrab.logrhythm.data.model

/**
 * Bristol stool scale 1..7 with plain-English aliases.
 * Values mirror phone.jsx BRISTOL exactly.
 */
data class BristolType(
    val n: Int,
    val plain: String,
    val description: String,
)

val BRISTOL_TYPES = listOf(
    BristolType(1, "Hard lumps", "Separate hard lumps, like nuts"),
    BristolType(2, "Lumpy",      "Sausage-shaped but lumpy"),
    BristolType(3, "Cracked",    "Sausage with cracks on its surface"),
    BristolType(4, "Smooth",     "Smooth and soft, like a sausage"),
    BristolType(5, "Soft lumps", "Soft blobs with clear-cut edges"),
    BristolType(6, "Mushy",      "Fluffy pieces, ragged edges"),
    BristolType(7, "Liquid",     "Entirely liquid, no solid pieces"),
)

fun bristol(n: Int): BristolType = BRISTOL_TYPES.first { it.n == n }

File: webapp/src/lib/bristol.ts
// Bristol stool scale 1..7 with plain-English aliases.
// Mirrors data/model/Bristol.kt and phone.jsx BRISTOL.

export interface BristolType {
  n: number
  plain: string
  description: string
}

export const BRISTOL_TYPES: BristolType[] = [
  { n: 1, plain: 'Hard lumps', description: 'Separate hard lumps, like nuts' },
  { n: 2, plain: 'Lumpy', description: 'Sausage-shaped but lumpy' },
  { n: 3, plain: 'Cracked', description: 'Sausage with cracks on its surface' },
  { n: 4, plain: 'Smooth', description: 'Smooth and soft, like a sausage' },
  { n: 5, plain: 'Soft lumps', description: 'Soft blobs with clear-cut edges' },
  { n: 6, plain: 'Mushy', description: 'Fluffy pieces, ragged edges' },
  { n: 7, plain: 'Liquid', description: 'Entirely liquid, no solid pieces' },
]

export function bristol(n: number): BristolType | undefined {
  return BRISTOL_TYPES.find((b) => b.n === n)
}
